diff --git a/kate/katetabbutton.cpp b/kate/katetabbutton.cpp
index dd999df34..140bb5675 100644
--- a/kate/katetabbutton.cpp
+++ b/kate/katetabbutton.cpp
@@ -44,6 +44,8 @@ TabCloseButton::TabCloseButton(QWidget * parent)
 
     // closing a tab closes the document
     setToolTip(i18n("Close Document"));
+
+    hide();
 }
 
 void TabCloseButton::paintEvent(QPaintEvent *event)
@@ -156,14 +158,14 @@ void KateTabButton::paintEvent(QPaintEvent *ev)
     if (! icon().isNull()) {
         const int y = (height() - 16) / 2;
         icon().paint(&p, margin, y, 16, 16);
-        leftMargin += 16;
+        //leftMargin += 16;
         leftMargin += margin;
     }
 
     // the width of the text is reduced by the close button + 2 * margin
-    const int w = width() // width of widget
-                - m_closeButton->width() - 2 * margin // close button
-                - leftMargin; // modified button
+    const int w = width() - 12; // width of widget
+                //- m_closeButton->width() - 2 * margin // close button
+                //- leftMargin; // modified button
 
     // draw text, we need to elide to xxx...xxx is too long
     const QString elidedText = QFontMetrics(font()).elidedText (text(), Qt::ElideMiddle, w);
